### PR TITLE
[SR-12912] Fix crash in test targets with Bundle.module in 5.3

### PR DESCRIPTION
**Pull request by MaxDesiatov**
Same as #2817, resolves SR-12912 in the `release/5.3` branch.

Setting the path for Bundle.module fails in the case where the bundle is accessed during tests.
This uses the already derived bundlePath instead.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci please test

**Comment by abertelrud** [comment]
Looks good to me, though I don't know where we are with being to get things into the SwiftPM 5.3 branch right now.

**Review by abertelrud — APPROVED** [review]

**Comment by MaxDesiatov** [comment]
Should I ping Rick Ballard to clarify this?

**Comment by abertelrud** [comment]
That sounds good; yes, @rballard  is release manager for 5.3, with @tomerd for future releases.

**Comment by rballard** [comment]
I saw the PR! I'm discussing schedules with some other Swift folks and will get back to you soon.

**Comment by rballard** [comment]
While this is an important fix, at this point in the schedule we're going to hold it; but let's keep the PR open and take it at the first post-5.3 opportunity.

**Comment by ffried** [comment]
Thanks everyone!

Please correct me if I'm wrong, but this basically renders SPM resources unusable for the following use cases:
- Test targets with resources.
- Normal (library) targets with resources that have unit tests.

The former is less critical IMHO. I also found a workaround for that and posted it [in the forums](https://forums.swift.org/t/swift-5-3-spm-resources-in-tests-uses-wrong-bundle-path/37051/10?u=ffried).

I find it rather difficult, however, to find a workaround for the latter case, since I'd like to avoid having to differentiate how resources are loaded in release vs. debug builds, just so that unit tests don't crash. Also, this would mean that the production code of loading resources is never actually tested.

**Comment by chrisballinger** [comment]
If we missed the mark for Xcode 12 GM, I would still really like to see this merged and available for an upcoming Xcode 12 point release, because otherwise it means we can't release SPM support for a popular library that includes resources (Core Data model definitions) and unit test coverage that involves those models: https://github.com/robbiehanson/XMPPFramework/pull/1177

What I'm not quite sure about is if this is merged into a Swift 5.3.1 release, our `Package@swift-5.3.swift` will remain broken on Swift 5.3.0. Is there a way to specify `Package@swift-5.3.1.swift`, or will library authors with resources and test coverage need to wait for Swift 5.4 before releasing SPM support?

**Comment by 3a4oT** [comment]
Please ship it with 5.3.

**Comment by rballard** [comment]
At this point in the schedule we need to hold this fix, but I expect us to take it ASAP for the first dot release post-5.3. This is a good question @chrisballinger re: whether we need to define a new package version for compatibility here; I'll chat with @neonichu about that.

**Comment by karimhm** [comment]
Will this fix [SR-13560](https://bugs.swift.org/browse/SR-13560) ?

**Review by airspeedswift — APPROVED** [review]

**Comment by rballard** [comment]
We've shipped 5.3 and should get this into the next patch release; approved to merge now. (I see @airspeedswift just beat me to it).

I chatted with @neonichu and we agreed that no new format version was needed here, as clients of a package don't run that package's tests, so clients on stock 5.3 shouldn't break when packages they use adopt resources.

**Review by rballard — APPROVED** [review]

**Comment by neonichu** [comment]
@karimhm I just double-checked and it does not fix this, SR-13560 seems like a separate issue.

**Pull request merged**
